staging/src/k8s.io/kubectl/pkg/polymorphichelpers/logsforobject_test.go

```diff
@@ -139,6 +139,66 @@ func TestLogsForObject(t *testing.T) {
 				FieldPath:  fmt.Sprintf("spec.containers{%s}", testPodWithOneContainers().Spec.Containers[0].Name),
 			}},
 		},
+		{
+			name: "pods list logs: different containers",
+			obj: &corev1.PodList{
+				Items: []corev1.Pod{
+					{
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "pod",
+							APIVersion: "v1",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "foo",
+							Namespace: "test",
+							Labels:    map[string]string{"test": "logs"},
+						},
+						Spec: corev1.PodSpec{
+							RestartPolicy: corev1.RestartPolicyAlways,
+							DNSPolicy:     corev1.DNSClusterFirst,
+							Containers: []corev1.Container{
+								{Name: "c1"},
+							},
+						},
```

Comment:
The test case should be annotated with `kubectl.kubernetes.io/default-container` to reproduce the issue.

The opts.Container names of different pods after code modification need to be compared so that people can know what the purpose of this test is.

I don't see that in the test code.

Reply:
thanks, updated.